common/test/TestDevice.cs

```diff
@@ -32,19 +32,21 @@ public class TestDevice
 
         private Device _device;
         private Client.IAuthenticationMethod _authenticationMethod;
+        private string _iotHubConnectionString;
 
-        private TestDevice(Device device, Client.IAuthenticationMethod authenticationMethod)
+        private TestDevice(Device device, Client.IAuthenticationMethod authenticationMethod, string iotHubConnectionString)
         {
             _device = device;
             _authenticationMethod = authenticationMethod;
+            _iotHubConnectionString = iotHubConnectionString;
         }
 
         /// <summary>
         /// Factory method.
         /// </summary>
         /// <param name="namePrefix"></param>
         /// <param name="type"></param>
-        public static async Task<TestDevice> GetTestDeviceAsync(string namePrefix, TestDeviceType type = TestDeviceType.Sasl)
+        public static async Task<TestDevice> GetTestDeviceAsync(string namePrefix, TestDeviceType type = TestDeviceType.Sasl, bool usePrimaryHub = true)
```

Comment:
What does it mean to use "primary hub"? We should only test against a single hub.

Reply:
This was to test if using multiple hubs brings down the run time (with current implementation). It was not found to be beneficial. This has been reverted.